controllers/users.controller.js

```diff
@@ -90,20 +85,129 @@ const updateUserDetails = async (req, res) => {
 				lastname: user.lastname,
 				userId: user._id,
 			},
-			success: true,
 		});
 	} catch (error) {
 		res.status(500).json({
-			success: false,
 			message: 'Request failed please check errorMessage key for more details',
 			errorMessage: error.message,
 		});
 	}
 };
 
+const getUserByIdFromDb = async (req, res) => {
+	try {
+		const { userId } = req.params;
+		console.log(userId);
+		const user = await User.findById({ _id: userId });
+
+		if (!user) {
+			res.status(404).json({ message: 'email does not exist!' });
+			return;
+		}
+
+		res.status(200).json({
+			response: {
+				email: user.email,
+				firstname: user.firstname,
+				lastname: user.lastname,
+				userId: user._id,
+			},
+		});
+	} catch (error) {
+		res.status(500).json({
+			message: 'Request failed please check errorMessage key for more details',
+			errorMessage: error.message,
+		});
+	}
+};
+
+const getNotesOfVideoOfUserFromDb = async (req, res) => {
+	try {
+		const { userId, videoId } = req.params;
+		const notes = await Note.find({ userId: userId, videoId: videoId });
+		res.status(200).json({ response: notes });
+	} catch (error) {
+		res.status(500).json({
+			message: 'Something went wrong',
+			errorMessage: error.message,
+		});
+	}
+};
+
+const getOrCreatePlaylistsOfUser = async (req, res) => {
```

Comment:
Okay, I have a suggestion here. And this is totally debatable. What I usually do is, I send out hardcoded values if there are no playlists of user. And when the user actually tries to add something to the playlist, then I create that playlist. I want to know your views on this. :D